File: tang-tool/tang-generator/src/main/java/com/tang/generator/enumeration/OrmType.java
package com.tang.generator.enumeration;

import java.util.Arrays;

/**
 * ORM 类型
 *
 * @author Tang
 */
public enum OrmType {

    MYBATIS("MyBatis"),

    MYBATIS_PLUS("MyBatis-Plus");

    private final String name;

    OrmType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static OrmType getOrmType(String name) {
        return Arrays.stream(OrmType.values())
            .filter(ormType -> ormType.getName().equals(name))
            .findFirst()
            .orElse(null);
    }

}

File: tang-tool/tang-generator/src/main/kotlin/com/tang/generator/enumeration/OrmType.kt
package com.tang.generator.enumeration

/**
 * ORM 类型
 *
 * @author Tang
 */
enum class OrmType(val value: String) {

    MYBATIS("MyBatis"),

    MYBATIS_FLEX("MyBatis-Flex"),

    MYBATIS_PLUS("MyBatis-Plus");

    companion object {
        @JvmStatic
        fun getOrmType(value: String): OrmType? {
            return entries.find { it.value == value }
        }
    }

}
